pkg/providers/cloudstack/validator.go

```diff
@@ -0,0 +1,197 @@
+package cloudstack
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const controlEndpointDefaultPort = "6443"
+
+type Validator struct {
+	cmk            ProviderCmkClient
+	machineConfigs map[string]*anywherev1.CloudStackMachineConfig
+}
+
+func NewValidator(cmk ProviderCmkClient, machineConfigs map[string]*anywherev1.CloudStackMachineConfig) *Validator {
+	return &Validator{
+		cmk:            cmk,
+		machineConfigs: machineConfigs,
+	}
+}
+
+type ProviderCmkClient interface {
```

Comment:
Let's create a `cloudstack.go` and keep the interface there even if that file doesn't have any other content right now. That would allow us to not have to move it back in a later PR. Same with any other constants or methods that would reside in that file.

Reply:
I'm not sure I agree. At this point, the provider itself doesn't need to interact with Cmk - its only interactions are for validations, which should presumably all occur in validator.go, so I'm inclined to keep the CloudMonkey structs isolated to the validator if possible. What do you think?

Reply:
Ohh I didn't realize that the provider doesn't need the cmk, which if that is the case you can disregard my comment. I was relating it to govc